Mongoid 7 support

grzegorz-jakubiak opened the pull request:
Addresses https://github.com/activeadmin/activeadmin-mongoid/issues/142

deivid-rodriguez commented on activeadmin-mongoid.gemspec:
```diff
@@ -17,10 +17,10 @@ Gem::Specification.new do |gem|
   gem.required_ruby_version = '>= 2.3'
 
-  gem.add_runtime_dependency 'mongoid',     ['~> 6.0']
+  gem.add_runtime_dependency 'mongoid',     '>= 6.0', '<= 7.0'
```
I think this is only going to allow mongoid 7.0.0, but not higher versions, and I don't think that's what you want? I think what you want is

```suggestion
  gem.add_runtime_dependency 'mongoid',     '>= 6.0', '< 8.a'
```

which allows any mongoid 7 version, but disallows any moingoid 8 version, including potential prereleases:

### Before

```
irb(main):009:0> Gem::Requirement.new(">= 6.0", "<= 7.0").satisfied_by?(Gem::Version.new("7.0.0"))
=> true
irb(main):010:0> Gem::Requirement.new(">= 6.0", "<= 7.0").satisfied_by?(Gem::Version.new("7.0.1"))
=> false
```

### After

```
irb(main):003:0> Gem::Requirement.new(">= 6.0", "< 8.a").satisfied_by?(Gem::Version.new("7.0.0"))
=> true
irb(main):004:0> Gem::Requirement.new(">= 6.0", "< 8.a").satisfied_by?(Gem::Version.new("7.0.1"))
=> true
```

grzegorz-jakubiak commented:
@deivid-rodriguez do you have any idea how to fix this issue?
```
NameError:
  uninitialized constant Responders::FlashResponder::ActionView
# ./vendor/bundle/ruby/2.4.0/gems/responders-2.4.1/lib/responders/flash_responder.rb:93:in `<module:FlashResponder>'
# ./vendor/bundle/ruby/2.4.0/gems/responders-2.4.1/lib/responders/flash_responder.rb:85:in `<module:Responders>'
# ./vendor/bundle/ruby/2.4.0/gems/responders-2.4.1/lib/responders/flash_responder.rb:1:in `<top (required)>'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources/responder.rb:3:in `require'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources/responder.rb:3:in `<class:Responder>'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources/responder.rb:2:in `<module:InheritedResources>'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources/responder.rb:1:in `<top (required)>'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources.rb:6:in `require'
# ./vendor/bundle/ruby/2.4.0/gems/inherited_resources-1.11.0/lib/inherited_resources.rb:6:in `<top (required)>'
# ./vendor/bundle/ruby/2.4.0/gems/activeadmin-2.6.0/lib/active_admin.rb:9:in `require'
# ./vendor/bundle/ruby/2.4.0/gems/activeadmin-2.6.0/lib/active_admin.rb:9:in `<top (required)>'
# ./lib/active_admin/mongoid.rb:6:in `require'
# ./lib/active_admin/mongoid.rb:6:in `<top (required)>'
# ./lib/activeadmin-mongoid.rb:1:in `require'
# ./lib/activeadmin-mongoid.rb:1:in `<top (required)>'
# /home/travis/.rvm/gems/ruby-2.4.6/gems/bundler-2.1.4/lib/bundler.rb:174:in `require'
# ./spec/spec_helper.rb:14:in `<top (required)>'
# ./spec/features/smoke_spec.rb:2:in `require'
# ./spec/features/smoke_spec.rb:2:in `<top (required)>'
```
I only noticed that `acitveadmin` has changed its dependency on `inherited_resources` to a newer version but I don't know the implications of that.

deivid-rodriguez commented:
The error means that `action_view` has not been require at some point. It sounds to me that it's the `responders` gem that should be explicitly depending on and requiring `action_view`, but I guess you can workaround the problem by explicitly requiring it from `activeadmin-mongoid`'s test suite. But honestly, I would actually remove the `Bundler.require' call that's failing in `spec/spec_helper.rb:14`, since I don't think it's required (if you run your test suite through `bundler`, I think the bundle shoud've been already setup and required at that point).

grzegorz-jakubiak commented:
Thanks for the suggestion but the error must be somewhere else, unfortunately, cause removing the line didn't help and the issue persists.

deivid-rodriguez commented:
Yeah, I figured that would simply move the error to somewhere else. Now try adding `require "action_view/railtie"` to `test_app/config/application.rb`. I will most likely fix the error. Still, we should probably report the missing dependency to `responders` anyways.

grzegorz-jakubiak commented:
didn't work 😢 we should definitely report it

deivid-rodriguez commented:
Oh, but you haven't removed the `Bundler.require` where the error is happening, can you try that?

grzegorz-jakubiak commented:
@deivid-rodriguez I fixed all the issues and pushed test fixes on to master and rebased this branch. When working on fixing mongoid issues I had a look at mongoid docs, https://github.com/sferik/rails_admin/commit/9ef623f6cba73adbf86833d9eb07f1be3924a133 and https://github.com/SechinAndrey/activeadmin-mongoid/commit/0fb31c9967da3dae32040f9b796e1d35ec14654b. Please have a second look at this PR. 😄

grzegorz-jakubiak commented:
@boie0025 have a look as well

deivid-rodriguez commented:
@grzegorz-jakubiak I'm sorry I'm not able to find time to review the specific mongoid fixes right now. But you got CI passing so that's a very good sign.

MiguelSavignano commented:
Hi @grzegorz-jakubiak 
I installed activeadmin-mongoid using this branch and evething it's ok in Rails 6.
